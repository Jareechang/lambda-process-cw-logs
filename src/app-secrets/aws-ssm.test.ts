import AWS from 'aws-sdk';

import {
    mapSecretNames,
    getAppSecrets
} from './aws-ssm';

jest.mock('aws-sdk');

import * as mockData from './samples';

describe('AWS ssm app wrapper', () => {
    const getParametersByPathMock = {
        promise: async() => {
            return mockData.getParametersByPath
        }
    };

    beforeEach(() => {
        // @ts-ignore
        AWS.SSM.mockImplementation(() => {
            return {
                getParametersByPath: () => getParametersByPathMock
            };
        });
    });

    it('should fetch the proper secret', async() => {
        const secrets = await getAppSecrets('/dev/application/slack_webhook/url');
        expect(secrets).toEqual({
            slack_url: mockData.getParametersByPath.Parameters[0].Value
        });
    });
});

describe('mapSecretNames()', () => {
    it('should properly return the correct mapping for secrets', () => {
        expect(mapSecretNames(
            mockData.getParametersByPath
        )).toMatchSnapshot();
    });
});
