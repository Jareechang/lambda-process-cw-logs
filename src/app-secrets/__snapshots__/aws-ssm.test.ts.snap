// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapSecretNames() should properly return the correct mapping for secrets 1`] = `
Object {
  "slack_url": "https://hooks.slack.com/services/<omitted>",
}
`;
